lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -842,14 +842,24 @@ def install_puppetlabs_release_repo_on( hosts, repo = nil, opts = options )
             opts = FOSS_DEFAULT_DOWNLOAD_URLS.merge(opts)
 
             case variant
-            when /^(fedora|el|centos|sles|cisco)$/
+            when /^(fedora|el|centos|sles|cisco_nexus|cisco_ios_xr)$/
               variant_url_value = (($1 == 'centos') ? 'el' : $1)
-              variant_url_value = 'cisco-wrlinux' if variant == 'cisco'
+              if variant == 'cisco_nexus'
+                variant_url_value = 'cisco-wrlinux'
+                version = '5'
+              end
+              if variant == 'cisco_ios_xr'
+                variant_url_value = 'cisco-wrlinux'
+                version = '7'
+              end
               remote = "%s/puppetlabs-release%s-%s-%s.noarch.rpm" %
                 [opts[:release_yum_repo_url], repo_name, variant_url_value, version]
 
-              if variant == 'cisco'
-                # cisco requires using yum to install the repo
+              if variant == 'cisco_ios_xr'
```

Comment:
Because on cisco_ios_xr, we have to do a localinstall of the repo because there are no repos configured at that moment on the VM.